CAP/gap/DerivedMethods.gi

```diff
@@ -3329,3 +3329,159 @@ AddFinalDerivation( IsEqualForCacheForMorphisms,
                     [ IsEqualForMorphisms ],
                     
   IsIdenticalObj : Description := "Use IsIdenticalObj for IsEqualForCacheForMorphisms" );
+
+## Final methods for BasisOfExternalHom & CoefficientsOfMorphism
+
+##
+AddFinalDerivation( BasisOfExternalHom,
+                    [
+                      [ HomomorphismStructureOnObjects ],
+                      [ HomomorphismStructureOnMorphismsWithGivenObjects ],
+                      [ InterpretMorphismFromDistinguishedObjectToHomomorphismStructureAsMorphism ],
+                      [ InterpretMorphismAsMorphismFromDistinguishedObjectToHomomorphismStructure ],
+                      [ DistinguishedObjectOfHomomorphismStructure ],
+                      [ MultiplyWithElementOfCommutativeRingForMorphisms ]
+                    ],
+                    [
+                      BasisOfExternalHom,
+                      CoefficientsOfMorphismWithGivenBasisOfExternalHom
+                    ],
+  function( a, b )
+    local hom_a_b, D, B, ConditionsListComplete;
+    
+    hom_a_b := HomomorphismStructureOnObjects( a, b );
+    
+    D := DistinguishedObjectOfHomomorphismStructure( CapCategory( a ) );
+    
+    B := ValueGlobal( "BasisOfExternalHom" )( D, hom_a_b );
```

Comment:
Why did you use this "trick" instead of calling
`BasisOfExternalHom( D, hom_a_b )`
?

Reply:
Sanity checks for the final derivations thinks this "BasisOfExternalHom" is being used in the category for which we derive the method. Hence, insists that it should be in the list of required methods.